File: algorithms/easy/neither_minimum_nor_maximum.ex
defmodule Solution do
  @spec find_non_min_or_max(nums :: [integer]) :: integer
  def find_non_min_or_max(nums) do
    nums = Enum.take(nums, 3)

    if length(nums) < 3 do
      -1
    else
      [_, a, _] = Enum.sort(nums)
      a
    end
  end
end

File: algorithms/easy/neither_minimum_nor_maximum.rs
impl Solution {
    pub fn find_non_min_or_max(nums: Vec<i32>) -> i32 {
        if nums.len() <= 2 {
            return -1;
        }
        let mut arr = &mut nums[..3].to_vec();
        arr.sort();
        arr[1]
    }
}
